kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/HasMetadataOperation.java

```diff
@@ -211,4 +211,20 @@ protected T periodicWatchUntilReady(int i, long started, long interval, long amo
       return periodicWatchUntilReady(i - 1, started, next, amount);
     }
   }
+
+  @Override
+  public T waitUntilReady(long amount, TimeUnit timeUnit) throws InterruptedException {
+    if (Readiness.isReadinessApplicable(getType())) {
+      long started = System.currentTimeMillis();
+      waitUntilExists(timeUnit.toMillis(amount));
+      long alreadySpent = System.currentTimeMillis() - started;
+
+      // if awaiting existence took very long, let's give at least 10 seconds to awaiting readiness
+      long remaining = Math.max(10_000, timeUnit.toMillis(amount) - alreadySpent);
```

Comment:
I am not sure if I fancy giving it extra time. Some possible alterations:

- subtract that 10 seconds from the waitUntilExists time.
- check if remaining is less or equal to zero and then just check if resource is ready.